src/storage/accessors/FileDataAccessor.ts

```diff
@@ -79,11 +80,13 @@ export class FileDataAccessor implements DataAccessor {
    */
   public async writeDocument(identifier: ResourceIdentifier, data: Readable, metadata: RepresentationMetadata):
   Promise<void> {
-    const link = await this.resourceMapper
-      .mapUrlToFilePath(identifier, metadata.contentType);
-    if (this.isMetadataPath(link.filePath)) {
+    if (this.isMetadataPath(identifier.path)) {
       throw new ConflictHttpError('Not allowed to create files with the metadata extension.');
     }
+    const link = await this.resourceMapper.mapUrlToFilePath(identifier, metadata.contentType);
+
+    // Check if we already have a corresponding file with a different extension
+    await this.verifyExistingExtension(link);
```

Comment:
Just wondering if there are cases where something could go wrong inbetween this and the writing of the file.
The chance this could happen is probably very small, but in case it could, should we somehow move this verification (and removal) to _after_ the writing of the file?
It's better to have two files after a crash than none at all.

Reply:
True, there's something to be said for that. The mapper is going to start crashing (or behaving weirdly, no idea) once there are two such files though but at least you still have your data I guess?

Reply:
On second thought going to leave it as is since this would require more rewriting than I first thought. Can always be changed later if really required.